cmd/ingester/app/flags.go

```diff
@@ -106,14 +108,61 @@ func AddFlags(flagSet *flag.FlagSet) {
 		DefaultDeadlockInterval,
 		"Interval to check for deadlocks. If no messages gets processed in given time, ingester app will exit. Value of 0 disables deadlock check.")
 
+	// TODO: Remove deprecated flags after 1.11
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixBrokers,
+		DefaultBroker,
+		"Deprecated; The comma-separated list of kafka brokers. i.e. '127.0.0.1:9092,0.0.0:1234'")
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixTopic,
+		DefaultTopic,
+		"Deprecated; The name of the kafka topic to consume from")
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixGroupID,
+		DefaultGroupID,
+		"Deprecated; The Consumer Group that ingester will be consuming on behalf of")
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixEncoding,
+		DefaultEncoding,
+		fmt.Sprintf(`Deprecated; The encoding of spans ("%s") consumed from kafka`, strings.Join(kafka.AllEncodings, "\", \"")))
 }
 
 // InitFromViper initializes Builder with properties from viper
 func (o *Options) InitFromViper(v *viper.Viper) {
-	o.Brokers = strings.Split(v.GetString(KafkaConfigPrefix+SuffixBrokers), ",")
-	o.Topic = v.GetString(KafkaConfigPrefix + SuffixTopic)
-	o.GroupID = v.GetString(KafkaConfigPrefix + SuffixGroupID)
-	o.Encoding = v.GetString(KafkaConfigPrefix + SuffixEncoding)
+	if brokers := strings.Split(v.GetString(DeprecatedKafkaConfigPrefix+SuffixBrokers), ","); len(brokers) > 0 {
+		fmt.Printf("WARNING: found deprecated option %s, please use %s instead\n",
+			DeprecatedKafkaConfigPrefix+SuffixBrokers,
+			KafkaConsumerConfigPrefix+SuffixBrokers,
+		)
+		o.Brokers = brokers
+	}
+	if topic := v.GetString(DeprecatedKafkaConfigPrefix + SuffixTopic); topic != "" {
+		fmt.Printf("WARNING: found deprecated option %s, please use %s instead\n",
+			DeprecatedKafkaConfigPrefix+SuffixTopic,
+			KafkaConsumerConfigPrefix+SuffixTopic,
+		)
+		o.Topic = topic
+	}
+	if groupId := v.GetString(DeprecatedKafkaConfigPrefix + SuffixGroupID); groupId != "" {
+		fmt.Printf("WARNING: found deprecated option %s, please use %s instead\n",
+			DeprecatedKafkaConfigPrefix+SuffixGroupID,
+			KafkaConsumerConfigPrefix+SuffixGroupID,
+		)
+		o.GroupID = groupId
+	}
+	if encoding := v.GetString(DeprecatedKafkaConfigPrefix + SuffixEncoding); encoding != "" {
+		fmt.Printf("WARNING: found deprecated option %s, please use %s instead\n",
+			DeprecatedKafkaConfigPrefix+SuffixEncoding,
+			KafkaConsumerConfigPrefix+SuffixEncoding,
+		)
+		o.Encoding = encoding
+	}
+
+	o.Brokers = strings.Split(v.GetString(KafkaConsumerConfigPrefix+SuffixBrokers), ",")
+	o.Topic = v.GetString(KafkaConsumerConfigPrefix + SuffixTopic)
+	o.GroupID = v.GetString(KafkaConsumerConfigPrefix + SuffixGroupID)
+	o.Encoding = v.GetString(KafkaConsumerConfigPrefix + SuffixEncoding)
```

Comment:
if the new flags are not specified, wouldn't all these values become empty?

Reply:
I've rework the PR and added a test for that.
I wasn't sure which flags we should prefer if both are specified, currently the old flags will win over the new flags if both are specified (as per the test)